Robot2018/src/tests/scriptupload/scriptupload.py

```diff
@@ -1,28 +1,29 @@
 #nt setup
 from networktables import NetworkTables
+import __future__
 NetworkTables.initialize(server='roboRIO-199-FRC.local')
 prefs = NetworkTables.getTable("Preferences")
 
 go = False #true when file has been successfully read
-file = [] #an array of strings which represent each line of the file
-filename = input("File name: ") #the name of the file to read (user input)
+lines = [] #an array of strings which represent each line of the file
+filename = raw_input("File name: ") #the name of the file to read (user input)
+oneline = "" #the file as a single string
+retry = False
 
 #loops until a file is read into the file array
 while not go:
-    try:
-        if filename == "quit":
+    if retry:
+        filename = raw_input("Not found. Try another name (enter to quit): ")#retry, or quit (in case the file doesn't exist)
+        if filename == "":
             quit()
-            
+    try:
         with open(filename) as script:
-            file = script.readlines()
-            #balex's testing
-            for line in file:
-                print(repr(line))
-            
+            lines = script.readlines()
+            "".join(lines) 
         break
     except:
-        filename = input("Not found. Try another name (enter to quit): ") or "quit" #retry, or quit (in case the file doesn't exist)
-
+        retry = True
 #puts the string array
+prefs.putString("autoscripts", file)
 prefs.putStringArray("autoscripts", file)
```

Comment:
Why did you put the same value in the same key as 2 different types?

Reply:
I think this is a merge error...